tensorflow/core/grappler/optimizers/auto_mixed_precision.h

```diff
@@ -24,6 +24,13 @@ namespace grappler {
 
 enum class AutoMixedPrecisionMode { CUDA, MKL };
 
+//Getting FP16 supported devices for ROCm
+#if TENSORFLOW_USE_ROCM
+extern std::array<std::string, 3> FP16SupportedDevices;
```

Comment:
why do we need to have this declaration in the header file?

Reply:
I had it this way so I wouldn't have two definitions of the var in `auto_mixed_precision.cc` and `auto_mixed_precision_test.cc` . Although `kMinGPUArch` is defined both in `auto_mixed_precision.cc` and `auto_mixed_precision_test.cc`. So I can do it that way.